crates/router/src/services/authentication/blacklist.rs

```diff
@@ -66,19 +66,26 @@ pub async fn check_user_in_blacklist<A: AppStateInfo>(
         .map(|timestamp| timestamp.map_or(false, |timestamp| timestamp > token_issued_at))
 }
 
-pub async fn check_role_in_blacklist<A: AppStateInfo>(
+pub async fn check_user_and_role_in_blacklist<A: AppStateInfo>(
     state: &A,
+    user_id: &str,
     role_id: &str,
     token_expiry: u64,
 ) -> RouterResult<bool> {
-    let token = format!("{}{}", ROLE_BLACKLIST_PREFIX, role_id);
+    let user_token = format!("{}{}", USER_BLACKLIST_PREFIX, user_id);
+    let role_token = format!("{}{}", ROLE_BLACKLIST_PREFIX, role_id);
     let token_issued_at = expiry_to_i64(token_expiry - JWT_TOKEN_TIME_IN_SECS)?;
     let redis_conn = get_redis_connection(state)?;
-    redis_conn
-        .get_key::<Option<i64>>(token.as_str())
+    let timestamp = redis_conn
```

Comment:
Let's not call it timestamp. It is not storing timestamp, it is bool value.

Either store timestamp in this and then use it separately in another statement or remove this variable entirely.